### ISSUE TITLE
opam config breaks fish shell in ubuntu trusty

### ISSUE THREAD

**Issue by AllanDaemon** [ocaml/opam#2244]
the open configs for fish put on shell init script make the shell broken.

Using Ubuntu Trusty 14.04 fully updated.
Opam installed by the PPA provided by the official install page (package version: `1.2.2-1ppa1~trusty`)
```
opam --version
1.2.2
```
using fish shell by default (`fish, version 2.0.0`)

with no ~/.opam folder, after ran `opam init` and choosing yes to set up the fish configuration, or if you don't do that but setup later with opam config, the shells breaks. I mean, every time you start a new shell (`ctrl+alt+t`) the fish gives a lots of errors, most of them are something like it couldn't found sed, because /bin is not in path. Every command (except fish own commands) will fail producing a lots of errors in the same way.

Commenting the fish init shell from being executed from fish init or commenting the line that setups `PATH` in `~/.opam/opam-init/variables.fish` brings it back to normal, but you can't use a command installed by opam.
```
#set -gx PATH "/home/user/.opam/system/bin" "$PATH";
set -gx OCAML_TOPLEVEL_PATH "/home/user/.opam/system/lib/toplevel";
set -gx PERL5LIB "/home/user/.opam/system/lib/perl5:$PERL5LIB";
set -gx MANPATH "$MANPATH" "/home/user/.opam/system/man";
set -gx CAML_LD_LIBRARY_PATH "/home/user/.opam/system/lib/stublibs:/usr/lib/ocaml/stublibs";
```

If you just comment this `PATH` line in `variables.fish` and latter set up it manually with `set PATH /home/user/.opam/system/bin/ $PATH`, it works fine though. The only lines in `~/.config/fish/config.fish` is the one added by opam:
```
# OPAM configuration
. /home/user/.opam/opam-init/init.fish > /dev/null 2> /dev/null or true
```

This is what shows when you open a new shell:
```
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
fish: Unknown command “hostname”
/usr/share/fish/functions/fish_prompt.fish (line 6): hostname|cut -d . -f 1
                                                     ^
in command substitution
	called on line 9 of file “/usr/share/fish/functions/fish_prompt.fish”,

in function “fish_prompt”,
	called on standard input,

in command substitution
	called on standard input,

Command 'hostname' is available in '/bin/hostname'
The command could not be located because '/bin' is not included in the PATH environment variable.
hostname: command not found
fish: Unknown command “cut”
/usr/share/fish/functions/fish_prompt.fish (line 6): hostname|cut -d . -f 1
                                                              ^
in command substitution
	called on line 9 of file “/usr/share/fish/functions/fish_prompt.fish”,

in function “fish_prompt”,
	called on standard input,

in command substitution
	called on standard input,

Command 'cut' is available in '/usr/bin/cut'
The command could not be located because '/usr/bin' is not included in the PATH environment variable.
cut: command not found
fish: Unknown command “uname”
/usr/share/fish/functions/fish_prompt.fish (line 6): uname
                                                     ^
in command substitution
	called on line 7 of file “/usr/share/fish/functions/fish_prompt.fish”,

in . (source) call of file “/usr/share/fish/functions/prompt_pwd.fish”,
	called on line 6 of file “/usr/share/fish/functions/fish_prompt.fish”,

in command substitution
	called on line 6 of file “/usr/share/fish/functions/fish_prompt.fish”,

in command substitution
	called on line 36 of file “/usr/share/fish/functions/fish_prompt.fish”,

in function “fish_prompt”,
	called on standard input,

in command substitution
	called on standard input,

Command 'uname' is available in '/bin/uname'
The command could not be located because '/bin' is not included in the PATH environment variable.
uname: command not found
test: Missing argument at index 2
fish: Unknown command “sed”
/usr/share/fish/functions/prompt_pwd.fish (line 8): echo $PWD | sed -e "s|^$HOME|~|" -e 's-\([^/]\)[^/]*/-\1/-g'
                                                                ^
in function “prompt_pwd”,
	called on line 6 of file “/usr/share/fish/functions/fish_prompt.fish”,

in command substitution
	called on line 36 of file “/usr/share/fish/functions/fish_prompt.fish”,

in function “fish_prompt”,
	called on standard input,

in command substitution
	called on standard input,

Command 'sed' is available in '/bin/sed'
The command could not be located because '/bin' is not included in the PATH environment variable.
sed: command not found
user@ > 
```

Every command fails accusing it don't exists. Strangely, this is what the shell PATH is set, which seems to be correct:
```
user@ > echo $PATH
/home/user/.opam/system/bin /usr/local/sbin /usr/local/bin /usr/sbin /usr/bin /sbin /bin /usr/games /usr/local/games
fish: Unknown command “sed”
/usr/share/fish/functions/prompt_pwd.fish (line 8): echo $PWD | sed -e "s|^$HOME|~|" -e 's-\([^/]\)[^/]*/-\1/-g'
                                                                ^
in function “prompt_pwd”,
	called on line 6 of file “/usr/share/fish/functions/fish_prompt.fish”,

in command substitution
	called on line 36 of file “/usr/share/fish/functions/fish_prompt.fish”,

in function “fish_prompt”,
	called on standard input,

in command substitution
	called on standard input,

Command 'sed' is available in '/bin/sed'
The command could not be located because '/bin' is not included in the PATH environment variable.
sed: command not found
```

Using bash instead of fish works ok.

**Comment by AltGr** [comment]
Indeed, sorry for this issue, and thanks for the detailed report. This was fixed already since 1.2.2 though, see #2165 !

**Comment by roryokane** [comment]
> This was fixed already since 1.2.2 though, see #2165 !

Not quite – though #2169 was merged, it seems it was too late to get into 1.2.2. I just installed Opam 1.2.2, and it still has this problem. I guess we’ll have to wait for 1.2.3.

**Comment by agam** [comment]
Just installed `opam`  and _still_ had the same problem with my fish shell; the workaround of removing the quotes around `$PATH` within the `init.fish` script fixed things.

**Comment by skrat** [comment]
Same here, the fix from @agam does the trick. Could you please update the fish files?

**Comment by rjbou** [comment]
Last week we finally released opam 2.0.0, that include the fish fix.
You can find in this [post](https://discuss.ocaml.org/t/opam-2-0-experimental-ppas/2446) information about Ubuntu packaging.

**Comment by evanhackett** [comment]
I recently installed opam while using fish, and it did still have 1 problem: it broke my man command by setting $MANPATH, so now my `man` command can't find anything except ocaml related things.
